lib/src/component_declaration/component_type_checking.dart

```diff
@@ -230,20 +231,28 @@ bool isComponentOfType(ReactElement instance, dynamic typeAlias, {
     return false;
   }
 
+  var finalInstance = instance;
+  var instanceType = finalInstance.type;
```

Comment:
The finalInstance variable can be eliminated, right?

I'd like to see this cleaned up, even if it's just in a commit to the WIP branch after we merge this.